pkg/cmd/prerunner.go

```diff
@@ -164,6 +165,27 @@ func IsFlagRequired(flag *pflag.Flag) bool {
 	return len(annotations) == 1 && annotations[0] == "true"
 }
 
+// Authenticated provides PreRun operations for commands that require a logged-in Confluent Cloud user.
+func (r *PreRun) AuthenticatedWithAPIKey(command *AuthenticatedCLICommand) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := r.Anonymous(command.CLICommand, true)(cmd, args); err != nil {
+			return err
+		}
+
+		ctx := command.Config.Context()
+		if ctx == nil {
+			return errors.NewErrorWithSuggestions("no context found", "Create a context with `confluent context create --bootstrap <bootstrap-server> --api-key <api-key> --api-secret <api-secret>`.")
```

Comment:
I think this should be an example in the produce/consume commands.

The schema registry version in the PR description should probably be an example, too.